util/CamReader.py

```diff
@@ -0,0 +1,123 @@
+import random
+import string
+from PIL import Image
+import matplotlib.pyplot as plt
+from os import system
+from time import sleep
+import cv2 as cv
+import numpy as np
+from sklearn.cluster import KMeans
+
+
+def _open_camera():
+    """
+
+    :return: camera IO object
+    """
+    #  0 -> camera number, if external camera is installed this number needs to be changed
+    # Since a hardware can be only accessible via one user, I/O limitation
+
+    cap = cv.VideoCapture(0)
+    if not cap.isOpened():
+        print("Cannot open camera")
+    return cap
+
+
+def _close_camera(cap):
+    """
+    :return: void : close the camera IO
+    """
+    # When everything done, release the capture
+    cap.release()
+
+
+def _capture_image(cap, w=120, h=140):
+    """
+    :return: the PIL image of live camera feed
+    """
+    # while True:
+    # Capture frame-by-frame
+    ret, frame = cap.read()
+    # if frame is read correctly ret is True
+    if not ret:
+        print("Can't receive frame (stream end?). Exiting ...")
+
+    # rescaling image to lower dimension
+
+    frame = cv.resize(frame, [w, h])
+    # gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
+    # frame = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
+    # Display the resulting frame
+    # frame is the coloured image
+    #  gray is the grayscaled version
+    # gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)\
+    return frame
```

Comment:
call `_convert_cv2_to_pil` on `frame` before returning it.

Reply:
hmm, better it's called later on numpy image, suppose tomorrow if we want to rollback to numpy image somehow (or use it in the util `method _print_to_dom_color` )